### ISSUE TITLE
Failed to initialize application error when launching new project

### ISSUE THREAD

**Issue by xvcgreg** [enso-org/enso#7060]
### What type of issue is this?

Intermittent – Occurring irregularly

### Is this issue blocking you from using Enso?

- [X] Yes, I can't use Enso because of this issue.

### Is this a regression?

- [ ] Yes, previous version of Enso did not have this issue.

### What issue are you facing?

![image](https://github.com/enso-org/enso/assets/108271897/c6fe2d3c-287c-4ebd-a9b3-37ede1497b94)

Launched Enso, created project, spun the environment and launched this new project. After a while I was greeted with this error and CB couldn't load it's components.

### Expected behaviour

New project launching w/o problems

### Enso Version

2023.6.19 nightly

### Browser or standalone distribution

Standalone distribution (cloud project)

### Browser Version or standalone distribution

standalone

### Operating System

Windows

### Operating System Version

Win11pro 22H2 22621.1555

### Hardware you are using

12th Gen Intel(R) Core(TM) i9-12900HK / RTX3060 Laptop / Nvidia Drivers 531.68

**Comment by indiv0** [comment]
Unable to reproduce when opening a new project on the latest nightly:

<img width="1391" alt="image" src="https://github.com/enso-org/enso/assets/641547/42c23b60-10c9-4e46-8ad2-b167d9450b1a">

**Comment by indiv0** [comment]
Just tried it again and still works for me. This time the nightly version is included in the screenshot, and it matches:
<img width="1381" alt="image" src="https://github.com/enso-org/enso/assets/641547/7c66ec5a-fdac-4f40-804a-e68b63ee93e0">

**Comment by wdanilo** [comment]
Another issue is that the error has 2 dots at the end of the sentence.

**Comment by xvcgreg** [comment]
> Another issue is that the error has 2 dots at the end of the sentence.

reported: #7068

**Comment by xvcgreg** [comment]
Updated the title, was able to reproduce it on local env:
![image](https://github.com/enso-org/enso/assets/108271897/8c0db2c0-603d-4f9b-9dad-324c4ad0d200)

**Comment by farmaazon** [comment]
> Updated the title, was able to reproduce it on local env:

This is a different error: In the description we haven't connected to the LS forbidding us any meaningful work, while in the comment we get "execution failed" meaning we connected successfully, but execution failed (and perhaps you are able to add nodes etc.).

**Comment by xvcgreg** [comment]
@farmaazon yes I'm able to add nodes
![image](https://github.com/enso-org/enso/assets/108271897/dd2f5d03-e7e4-4a27-815e-12be64f87e43)

**Comment by farmaazon** [comment]
> @farmaazon yes I'm able to add nodes

Again, _this is a different issue_. Please mind the difference in error message. In the original screenshot, there is "Failed to initialize RPC connection", in your last comment it's "Execution failed"

**Comment by hubertp** [comment]
I'm also unable to reproduce this (`Failed to initialize RPC connection`) problem. 

But I suppose I could we could create an artificial setup where it does happen. When we fail to establish the most basic stuff, which can still happen in the worst case, then we should still prepare for this. I guess the only possibility then is to just either restart automatically or request the user to do that (preferably via a slightly more informed message)?

**Comment by farmaazon** [comment]
> But I suppose I could we could create an artificial setup where it does happen. 

That indeed would be helpful for testing.

>  I guess the only possibility then is to just either restart automatically or request the user to do that (preferably via a slightly more informed message)?

Even now we try to initialize IDE several times, the message suggests all of those failed. We would need to ask for reporting, similarly to the "Execution failed" message.

**Comment by xvcgreg** [comment]
This issue is blocking #7023 can we push it forward @hubertp ?

**Comment by hubertp** [comment]
I fail to see how this is related to #7023 - the original error, which btw no one is able to reproduce under regular circumstances, has nothing to do with random console warnings that are mentioned in that PR.

It can also be the case that I'm not seeing some implicit connection but then @xvcgreg @indiv0, please be more specific.

**Comment by xvcgreg** [comment]
@hubertp it was mentioned here: https://github.com/enso-org/enso/pull/7023#issuecomment-1598707245

@indiv0 please fill us in with a bit more information

**Comment by jdunkerley** [comment]
Running on a virtual machine with no graphics card, this happens each time.
The IDE times out while starting up the blank project.

![image](https://github.com/enso-org/enso/assets/4699705/9f942e6e-32ab-47db-ac88-5ad8a5c45013)

I can happily run and gather any logs needed - this is trivial to repro on a Win10 Azure VM (4 core 16Gb).

**Comment by hubertp** [comment]
@farmaazon I guess IDE could attempt to retry after this and/or display some kind of information about (hopefully) temporary connection issues?

**Comment by farmaazon** [comment]
It is implemented, but somehow does not work as I see. So we have a regression in the IDE.

**Comment by xvcgreg** [comment]
> It is implemented, but somehow does not work as I see. So we have a regression in the IDE.

@farmaazon please update the labels
@jdunkerley could you gather those logs - I have tried to reproduce for the past 20mins and came back emptyhanded

**Comment by farmaazon** [comment]
> @farmaazon please update the labels

I will create a new issue instead. 

I want to note, that in the screenshot from the original description IDE actually tried to reconnect several times, and all attempts failed, because only after retrying we display "Failed to initialize application" message:
```rust
        match self.initialize_ide_controller_with_retries().await {
            Ok(controller) => {
                let can_manage_projects = controller.can_manage_projects();
                let ide = Ide::new(ensogl_app, view, controller);
                if can_manage_projects {
                    if let Some(project) = &self.config.project_to_open {
                        ide.open_or_create_project(project.clone());
                    }
                }
                info!("IDE was successfully initialized.");
                Ok(ide)
            }
            Err(error) => {
                let message = format!("Failed to initialize application: {error}");
                status_bar.add_event(ide_view::status_bar::event::Label::new(message));
                Err(FailedIde { view })
            }
        }
```

**Comment by xvcgreg** [comment]
> > @farmaazon please update the labels
> 
> I will create a new issue instead.

@farmaazon please link it here

**Comment by farmaazon** [comment]
This one: #7163

**Comment by JaroslavTulach** [comment]
Looks like #7163 is fixed. Is this issue still reproducible? Reopen, if so.
